### PR TITLE
Remove deprecated DISALLOW_COPY_AND_ASSIGN from Brave Ads

### PR DESCRIPTION
**Pull request by tmancey**
Resolves https://github.com/brave/brave-browser/issues/9165

## Submitter Checklist:

- [x] Submitted a [ticket](https://github.com/brave/brave-browser/issues) for my issue if one did not already exist.
- [x] Used Github [auto-closing keywords](https://help.github.com/articles/closing-issues-via-commit-messages/) in the commit message.
- [ ] Added/updated tests for this change (for new code or code which already has tests).
- Verified that these changes build without errors on
  - [ ] Android
  - [ ] iOS
  - [ ] Linux
  - [x] macOS
  - [ ] Windows
- Verified that these changes pass automated tests (unit, browser, security tests) on
  - [ ] iOS
  - [ ] Linux
  - [x] macOS
  - [ ] Windows
- [x] Verified that all lint errors/warnings are resolved (`npm run lint`)
- [x] Ran `git rebase master` (if needed).
- [x] Ran `git rebase -i` to squash commits (if needed).
- [x] Tagged reviewers and labelled the pull request as needed.
- [ ] Request a security/privacy review as needed.
- [x] Add appropriate QA labels (QA/Yes or QA/No) to include the closed issue in milestone
- [ ] Public documentation has been updated as necessary. For instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protections
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Custom-Headers
  - [ ] https://github.com/brave/brave-browser/wiki/Web-compatibility-issues-with-tracking-protection
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide

## Reviewer Checklist:

- [ ] New files have MPL-2.0 license header.
- [ ] Request a security/privacy review as needed.
- [ ] Adequate test coverage exists to prevent regressions 
- [ ] Verify test plan is specified in PR before merging to source

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on.
- [ ] All relevant documentation has been updated.

### CODE REVIEW

**Comment by tmancey** [comment]
@iefremov pointed me to https://groups.google.com/a/chromium.org/forum/#!msg/chromium-dev/BgFYNQnAUDk/slCE4eDIAwAJ

File: components/brave_ads/browser/ads_notification_handler.h [hunk 1]
```diff
@@ -80,7 +80,10 @@ class AdsNotificationHandler : public NotificationHandler {
    private:
     AdsNotificationHandler* const pointer_;
 
-    DISALLOW_COPY_AND_ASSIGN(UnownedPointer);
+    UnownedPointer(
```
Comment (iefremov) [inline components/brave_ads/browser/ads_notification_handler.h:83]: it looks like most of these fit into 80 chars, so could you please make these declarations one-liners?
Also they mostly should be present in `public:` section

Comment (tmancey) [reply components/brave_ads/browser/ads_notification_handler.h]: Fixed

**Comment by tmancey** [comment]
macOS failed CI, restarting

**Pull request merged**